client/src/CoqDocument.ts

```diff
@@ -316,7 +318,7 @@ export class CoqDocument implements vscode.Disposable {
       if(!newPath)
         return false;
       try {
-        return await fs.existsSync(path.join(newPath, 'coqtop')) || await fs.existsSync(path.join(newPath, 'coqtop.exe'))
+        return fs.existsSync(path.join(newPath, coqtopExe))
```

Comment:
This doesn't check `coqtopExe + '.exe'`.

Reply:
That is correct. If the user specifies a `newPath` via configuration, they'd be competent enough to specify a `coqtopExe` as well, no?

The VSCoq code is overcomplicated, and I intend to clean up this complexity along with my patches.

Reply:
> The VSCoq code is overcomplicated, and I intend to clean up this complexity along with my patches.

Matter of style and not my call, but separating logically distinct changes helps when bisecting. And a simpler patch would require less testing to merge.
But if one is cleaning this up, why are there *two* variants of the logic? It also seems, from reading a bit more code, that `coqtop.exe` is tried when inputting a new path, but not here? If they’re inconsistent for good reason, shouldn’t that be explained in a comment?

> If the user specifies a newPath via configuration, they'd be competent enough to specify a coqtopExe as well, no?

Having dealt with students, I think “competent user” is too restrictive.
There’s a million valid reasons why a competent user would assume otherwise — most Windows programs just infer `.exe`, so writing `coqtop` is absolutely correct.
Worse, the default setting of coqtopExe works despite the `.exe` mismatch, so I expect that changing the binary folder (which is orthogonal) wouldn’t break this. The description is now a bit better, but it’s clearly incorrect enough I’d file a bug.

Reply:
> most Windows programs just infer .exe, so writing coqtop is absolutely correct

Oh, then we need the logic. Sigh.